### PR TITLE
refactor memory logs

### PR DESCRIPTION
**Pull request by Zhangxunmt**
### Description
Refactor memory logs to 1) add more logs info from pen testers' suggestion and 2) replace conversation/interaction with memory/message in the logs based on suggestion from Tec doc writer. 
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: memory/src/main/java/org/opensearch/ml/memory/action/conversation/UpdateConversationTransportAction.java [hunk 1]
```diff
@@ -71,7 +71,7 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<Update
 
                 cmHandler.updateConversation(conversationId, updateContent, getUpdateResponseListener(conversationId, listener, context));
             } catch (Exception e) {
-                log.error("Failed to update Conversation " + conversationId, e);
+                log.error("Failed to update Memory " + conversationId, e);
```
Comment (dhrubo-os) [inline memory/src/main/java/org/opensearch/ml/memory/action/conversation/UpdateConversationTransportAction.java:74]: `Failed to update Memory with ID`

File: memory/src/main/java/org/opensearch/ml/memory/action/conversation/UpdateInteractionTransportAction.java [hunk 2]
```diff
@@ -69,7 +69,7 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<Update
 
             cmHandler.updateInteraction(interactionId, updateContent, getUpdateResponseListener(interactionId, listener, context));
         } catch (Exception e) {
-            log.error("Failed to update Interaction " + interactionId, e);
+            log.error("Failed to update Message " + interactionId, e);
```
Comment (dhrubo-os) [inline memory/src/main/java/org/opensearch/ml/memory/action/conversation/UpdateInteractionTransportAction.java:72]: with ID --> let's keep the message pattern consistent.

File: memory/src/main/java/org/opensearch/ml/memory/index/ConversationMetaIndex.java [hunk 3]
```diff
@@ -270,17 +272,17 @@ public void deleteConversation(String conversationId, ActionListener<Boolean> li
                             internalListener.onResponse(false);
                         }
                     }, e -> {
-                        log.error("Failure deleting conversation " + conversationId, e);
+                        log.error("Failure deleting memory " + conversationId, e);
```
Comment (dhrubo-os) [inline memory/src/main/java/org/opensearch/ml/memory/index/ConversationMetaIndex.java:275]: with ID

File: memory/src/main/java/org/opensearch/ml/memory/action/conversation/UpdateInteractionTransportAction.java (hunk 2 above)
Comment (Zhangxunmt) [reply memory/src/main/java/org/opensearch/ml/memory/action/conversation/UpdateInteractionTransportAction.java]: added them

**Review by b4sjoo — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]

File: memory/src/main/java/org/opensearch/ml/memory/action/conversation/CreateConversationTransportAction.java [hunk 4]
```diff
@@ -86,7 +86,7 @@ protected void doExecute(Task task, CreateConversationRequest request, ActionLis
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().newStoredContext(true)) {
             ActionListener<CreateConversationResponse> internalListener = ActionListener.runBefore(actionListener, () -> context.restore());
             ActionListener<String> al = ActionListener.wrap(r -> { internalListener.onResponse(new CreateConversationResponse(r)); }, e -> {
-                log.error("Failed to create new conversation with name " + request.getName(), e);
+                log.error("Failed to create new memory with name " + request.getName(), e);
```
Comment (samuel-oci) [inline memory/src/main/java/org/opensearch/ml/memory/action/conversation/CreateConversationTransportAction.java:89]: please use parameterized logging in Java.
`"Failed to create new memory with name {}",request.getName()`

File: memory/src/main/java/org/opensearch/ml/memory/action/conversation/CreateConversationTransportAction.java [hunk 5]
```diff
@@ -96,7 +96,7 @@ protected void doExecute(Task task, CreateConversationRequest request, ActionLis
                 cmHandler.createConversation(name, applicationType, al);
             }
         } catch (Exception e) {
-            log.error("Failed to create new conversation with name " + request.getName(), e);
+            log.error("Failed to create new memory with name " + request.getName(), e);
```
Comment (samuel-oci) [inline memory/src/main/java/org/opensearch/ml/memory/action/conversation/CreateConversationTransportAction.java:99]: same as before please use parameterized logging going forward

**Pull request merged**